go/src/koding/klient/machine/machinegroup/syncs/syncs_test.go

```diff
@@ -59,19 +55,15 @@ func TestSyncsDrop(t *testing.T) {
 
 	// Create new sync.
 	mountID := mount.MakeID()
-	s, err := syncs.New(syncs.SyncsOpts{
-		WorkDir:       wd,
-		NotifyBuilder: silent.SilentBuilder{},
-		SyncBuilder:   discard.DiscardBuilder{},
-	})
+	s, err := syncs.New(syncs.Options{WorkDir: wd})
 	if err != nil {
 		t.Fatalf("want err = nil; got %v", err)
 	}
 	defer s.Close()
 
-	if err := s.Add(mountID, m, func() (client.Client, error) {
-		return clienttest.NewClient(), nil
-	}); err != nil {
+	dynSSH, dynClient := dynFunctions()
+	nb, sb := silent.Builder{}, discard.Builder{}
```

Comment:
I know you like described approach because I saw it in your fuse impl. Can's see any advantages/disadvantages of this approach compared to raw type. 

`ioutil.Discard` must deal with platform-dependent code and it could be hard to implement this with `Discard`/`devNull` types directly.